servicetalk-http-netty/src/test/java/io/servicetalk/http/api/BaseContentCodingTest.java

```diff
@@ -0,0 +1,245 @@
+            this.list = list;
+        }
+    }
+
+    protected enum Client {
+        DEFAULT(emptyList()),
+        GZIP_ONLY(singletonList(gzipDefault())),
+        GZIP_ID(asList(gzipDefault(), identity())),
+        GZIP_DEFLATE_ID(asList(gzipDefault(), deflateDefault(), identity())),
+        ID_ONLY(singletonList(identity())),
+        ID_GZIP(asList(identity(), gzipDefault())),
+        ID_DEFLATE(asList(identity(), deflateDefault())),
+        ID_GZIP_DEFLATE(asList(identity(), gzipDefault(), deflateDefault())),
+        DEFLATE_ONLY(singletonList(deflateDefault())),
+        DEFLATE_ID(asList(deflateDefault(), identity()));
+
+        List<ContentCodec> list;
+
+        Client(List<ContentCodec> list) {
+            this.list = list;
+        }
+    }
+
+    protected enum Request {
+        ID(identity()),
+        GZIP(gzipDefault()),
+        DEFLATE(deflateDefault());
+
+        ContentCodec codec;
+
+        Request(ContentCodec codec) {
+            this.codec = codec;
+        }
+    }
+
+    protected enum Expectation {
+        SHOULD_PASS(true),
+        SHOULD_FAIL(false);
+
+        boolean valid;
+        Expectation(boolean valid) {
+            this.valid = valid;
+        }
+    }
+
+    protected enum Protocol {
+        H1(h1Default()),
+        H2(h2Default());
+
+        HttpProtocolConfig config;
+
+        Protocol(HttpProtocolConfig config) {
+            this.config = config;
+        }
+    }
+
+    protected static class Scenario {
+        final ContentCodec requestEncoding;
+        final List<ContentCodec> clientSupported;
+        final List<ContentCodec> serverSupported;
+        final HttpProtocolConfig protocol;
+        final boolean valid;
+        final boolean isH2;
+
+        Scenario(final ContentCodec requestEncoding,
+                 final List<ContentCodec> clientSupported, final List<ContentCodec> serverSupported,
+                 final Protocol protocol, final boolean valid) {
+            this.requestEncoding = requestEncoding;
+            this.clientSupported = clientSupported;
+            this.serverSupported = serverSupported;
+            this.valid = valid;
+            this.isH2 = protocol == H2;
+            this.protocol = isH2 ? h2Default() : h1Default();
+        }
+
+        static Scenario when(final Protocol protocol, final Server server,
+                             final Client client, final Request request, final Expectation expectation) {
+            return new Scenario(request.codec, client.list, server.list, protocol, expectation.valid);
+        }
+
+        @Override
+        public String toString() {
+            return "When a client that supports " + encString(clientSupported) + ", sends a request encoded with "
+                    + requestEncoding.name() + ", to an " + (isH2 ? "H2" : "H1") + " server that supports "
+                    + encString(serverSupported) + ", the request should be " + (valid ? "ACCEPTED" : "REJECTED");
```

Comment:
Consider simplified parametrized format:
```
client=[...], server=[...], request={}, protocol={}, shouldPass={}
```
Current format is hard to read in intellij idea and hard to run via command line a specific gradle test.

Reply:
Included index in the params (to allow running from cmd easier [1]), and simplified the text.

1. `./gradlew --info :servicetalk-http-netty:test --tests "io.servicetalk.http.netty.ServiceTalkContentCodingTest.testCompatibility[15*]"`